lib/Driver/ToolChains.cpp

```diff
@@ -1340,15 +1341,23 @@ toolchains::GenericUnix::constructInvocation(const LinkJobAction &job,
   return {"clang++", Arguments};
 }
 
-std::string toolchains::Cygwin::getDefaultLinker() const {
-  // Cygwin uses the default BFD linker, even on ARM.
-  return "";
+std::string
+toolchains::Android::getTargetForLinker() const {
+  // Explicitly set the linker target to "androideabi", as opposed to the
+  // llvm::Triple representation of "armv7-none-linux-android".
+  // This is the only ABI we currently support for Android.
```

Comment:
Let's actually assert that then. That way, we'll know very loudly when it changes.

```
assert(getTriple().getArch() == llvm::Triple::arm && "No other architectures supported for Android");
```

or something.